File: Data Structures & Algorithms/string-encode-and-decode/submission-0.go
type Solution struct{}

func (s *Solution) Encode(strs []string) string {
 var encoded string
 for _, val := range strs {
	encoded += strconv.Itoa(len(val))+"#"+val
 }
 return encoded
}

func (s *Solution) Decode(encoded string) []string {
 decoded := []string{}
 i := 0
 for i < len(encoded){
	j := i
	for encoded[j] != '#' {
		j++
	}
	charlen,_ := strconv.Atoi(encoded[i:j])
	i = j+1
	decoded = append(decoded, encoded[i:charlen+i])
	i += charlen
 }
 return decoded
}

File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        base = strs[0]

        for i in range(1,len(strs)):
            j = 0
            while j < min(len(strs[i]), len(base)):
                if base[j] != strs[i][j]:
                    break
                j+=1
            base = base[:j]

        return base
                